### ISSUE TITLE
Core: add pinning

### ISSUE THREAD

**Issue by GPMueller** [spirit-code/spirit#205]
Related to #104.
Four possibilities of how one might want to do this:
- add a (homogeneous?) background, which can also be reached on longer-range interactions
- add a single layer around everything
- inhomogeneous anisotropy distribution (set the border spins to have high anisotropy in any direction)
- exclude certain spins from updates (in this case the border spins)

This may require some refactoring...

**Comment by GPMueller** [comment]
Implemented basic possibility for pinning in 816a95ae77ad7b9a736d40d0fa553cc8ae0d2b95. The anisotropic magnetic field can be set to pin certain spins by setting a high field magnitude there.

**Comment by GPMueller** [comment]
The way to go is to re-set certain spins after each `Iteration` in the `Optimizer`, using data stored in `Parameters_Method`. The necessary data is:
- `pin_na_left`, `pin_na_right`, `pin_nb_left`, ... (how many layers of cells should be pinned on each side of a translation direction) or just `pin_na` etc.
- `pin_na_left_spins` containing direction and index inside the basis

e.g.:
```Python
pin_na_left 2
# ilayer  iatom  sx sy sz
0   0   0 0 1
0   1   1 0 0
1   0   0 0 1
1   1   1 0 0
pin_na_right 2
pin_nb_right 1
```
**OR** (far simpler):
```Python
pin_na 2
#  iatom  sx sy sz
 0   0 0 1
 1   1 0 0
```

**Comment by GPMueller** [comment]
Started working on this with fd655a6cebc2c984ec16890ff9edb1933412e977 on `feature-pinning`.
In order to implement this correctly, the LLG forces (temperature, stt, ...) need to be calculated in `Method_LLG`, not `Optimizer_SIB` and therefore the Optimizers need to be adapted to also support "damping-only". This is tracked in #182.

**Comment by GPMueller** [comment]
Pinning is working correctly since b98ba7966a3ba69a3c3d802ffc8ebe84976d768c on `feature-pinning`.

As it turns out it is not significantly better to do the pinning all inside `Method_LLG` and moving the forces there has other problems attached to it...

**Comment by GPMueller** [comment]
Merged into `develop` with 204e8542ae719c8cb9d8b3295a90c5cdaa66e551.
